Lib/test/test_dataclasses.py

```diff
@@ -3672,6 +3681,36 @@ def test_no_types(self):
                                              'y': int,
                                              'z': 'typing.Any'})
 
+    def test_module_attr(self):
+        self.assertEqual(ByMakeDataClass.__module__, __name__)
+        self.assertEqual(ByMakeDataClass(1).__module__, __name__)
+
```

Comment:
It seems worth also checking here that manually providing the module argument works:

```suggestion
self.assertEqual(WrongModuleMakeDataclass.__module__, "custom")
```